nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/core/storage/sql/aws/S3DirectBatchHandler.java

```diff
@@ -0,0 +1,329 @@
+        blobInfo.length = updatedObjectMetadata.getContentLength();
+        blobInfo.digest = updatedObjectMetadata.getContentMD5();
+        blobInfo.mimeType = mimeType;
+
+        Blob blob = new BinaryBlob(
+                new LazyBinary(blobInfo.key, transientStoreName, null)
+                , blobInfo.key
+                , blobInfo.filename
+                , blobInfo.mimeType
+                , blobInfo.encoding
+                , blobInfo.digest
+                , blobInfo.length
+        );
+
+        try {
+            batch.addFile(fileIndex, blob, blobInfo.filename, blobInfo.mimeType);
+        } catch (Exception e) {
+            throw new NuxeoException(e);
+        }
+
+        return true;
+    }
+
+    private boolean containsRequired(Map<String, String> configProperties) {
+        return configProperties.containsKey(Constants.S3TransientStoreConfig.Properties.AWS_SECRET_KEY_ID)
+                && configProperties.containsKey(Constants.S3TransientStoreConfig.Properties.AWS_SECRET_ACCESS_KEY)
+                && configProperties.containsKey(Constants.S3TransientStoreConfig.Properties.AWS_ROLE_ARN)
+                && configProperties.containsKey(Constants.S3TransientStoreConfig.Properties.AWS_REGION)
+                && configProperties.containsKey(Constants.S3TransientStoreConfig.Properties.AWS_BUCKET)
+                && configProperties.containsKey(Constants.S3TransientStoreConfig.Properties.TRANSIENT_STORE_NAME);
+    }
+
+    @Override
+    protected TransientStore getTransientStore() {
+        if (transientStore == null) {
+            TransientStoreService service = Framework.getService(TransientStoreService.class);
+            transientStore = service.getStore(transientStoreName);
+        }
+
+        return transientStore;
+    }
+
+    private AWSSecurityTokenService initClient(String awsSecretKeyId, String awsSecretAccessKey, String region) {
+        return AWSSecurityTokenServiceClientBuilder
+                .standard()
+                .withRegion(region)
+                .withCredentials(new AWSStaticCredentialsProvider(new BasicAWSCredentials(awsSecretKeyId, awsSecretAccessKey)))
+                .build()
+                ;
+    }
+
+    protected AmazonS3 initS3Client(String awsSecretKeyId, String awsSecretAccessKey, String region, boolean useAccelerate) {
+        AWSCredentialsProvider awsCredentialsProvider = getAwsCredentialsProvider(awsSecretKeyId, awsSecretAccessKey);
+        return AmazonS3ClientBuilder
+                .standard()
+                .withRegion(region)
+                .withCredentials(awsCredentialsProvider)
+                .withAccelerateModeEnabled(useAccelerate)
+                .build()
+                ;
+    }
+
+    protected AWSCredentialsProvider getAwsCredentialsProvider(String awsSecretKeyId, String awsSecretAccessKey) {
+        AWSCredentialsProvider result;
+        if (isBlank(awsSecretKeyId) || isBlank(awsSecretAccessKey)) {
```

Comment:
Please be consistent with `StringUtils#isEmpty`, `StringUtils#isNotEmpty`, `StringUtils#isBlank`
 and`StringUtils#isNotBlank`. Import them statically or not.

Reply:
Done